### PR TITLE
[kostalinverter] Fix for the Kostal inverter binding to work with different firmware releases regarded to  PIKO 10-20

### PR DESCRIPTION
**Pull request by basse04**
Changes to  check in:
	changed:        ../../../../../../../../../README.md
	changed:        SecondGenerationHandler.java
Signed-off-by: Örjan Backsell <orjan.backsell@gmail.com>

### CODE REVIEW

**Comment by lolodomo** [comment]
Please update your title to explain what this PR is fixing or enhancing.

File: bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java [hunk 1]
```diff
@@ -172,6 +172,9 @@ public void initialize() {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                         scheduleWithFixedDelayException.getClass().getName() + ":"
                                 + scheduleWithFixedDelayException.getMessage());
+            } catch (InterruptedException e) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java:175]: Please add that just as first instruction in the block:
```
Thread.currentThread().interrupt();
```

File: bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java [hunk 2]
```diff
@@ -172,6 +172,9 @@ public void initialize() {
...
+                logger.debug("Updating inverter status failed: ", e);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java:176]: Maybe a more appropriate message ?
Something like : "Refresh was interrupted"

File: bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java [hunk 3]
```diff
@@ -327,7 +334,7 @@ private void refresh() {
             }
         } catch (final RuntimeException e) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java:335]: In fact, `refresh` is called only by the scheduled job by initialize.
You have to choose where do you catch `RuntimeException`; doing it at both places has no real sense.
I would suggest to remove the exception handling here.

File: bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java [hunk 4]
```diff
@@ -347,6 +354,12 @@ private final void preSetExecuteConfigurationChanges(HttpClient httpClientHandle
     private final String callURL(String dxsEntriesCall, HttpClient httpClient) {
         String jsonDxsResponse = "";
         try {
+            if (!httpClient.isRunning()) {
+                try {
+                    httpClient.start();
+                } catch (Exception e1) {
+                }
+            }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java:362]: Never do that when the binding is using the common HTTP client (common to all bindings).
The common HTTP client will be already started when the binding is started.

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
Please also note you did not sign off properly your PR.

**Comment by basse04** [comment]
Hi, 
looks strange with 7 commits and it is, but I got a message that I wasn't allowed to log in a couple of times, do not know why, but suddenly it works and then maybe with all the earlier not allowed ones also sent.
Br Basse

**Comment by lolodomo** [comment]
Don't know what you did, but that PR is totally different to what it was yesterday. 
And you have now conflicts.

**Comment by basse04** [comment]
I had problems to get the changes uploaded, I did it from Eclipse and that had been a nightmare sometimes before also, probebly depending on me.
But what I can understand so are the changes now at basse04:kostalinverter_20220929, should I have make another branch like kostalinverter_20221003 and upload the changes to there?

**Comment by lolodomo** [comment]
You must keep the same branch. 
If not, a new PR will be created. 
So do it only if you cannot restore something clean in your current branch.

**Comment by basse04** [comment]
Thanks for your help, so you can use the changes I have uploaded today to basse04:kostalinverter_20220929 ? Otherwise, Please, just let me know.

**Comment by lolodomo** [comment]
What you have in your original branch is what we see now here. 
If you have a backup of your expected changes, you can remove your local branch, create it again with the same name, apply again your changes, commit and finally force push your branch.

**Comment by basse04** [comment]
Hi,
my best excuses. I was not able to save kostalinverter_20220929. I have totally meshed it up and I finally found the reason, I have updated it with an old branch 20220129 by mistake without to note that until just now. So I will close this PR and make a new one with the advices you have give me in 20220929.
Br Basse

**Pull request closed without merging**